agent/checks/alias.go

```diff
@@ -203,6 +229,8 @@ func (c *CheckAlias) processChecks(checks []*structs.HealthCheck) {
 		msg = "All checks passing."
 	}
 
+	// TODO(rb): if no matching checks found should this default to critical?
```

Comment:
It seems like it should. No matching checks means that there were no registered checks for the aliased service nor for the node.

Reply:
I left this as a drive-by comment mostly as an observation. I am not sure what the original intended operation of this default behavior was supposed to be and this surprised me.

Reply:
I'm curious if @banks has any thoughts on this specific point?

Reply:
I'd argue not. The current behaviour if you have a service registered with zero checks is that it *will* show up in DNS and service discovery results. If we made an alias Critical if the target service has no matching checks then the target would be considered passing while the alias would be critical which is not what I'd expect.

The overall goal is that the Alias check instance's status should always match the overall aggregate status of the targer service instance which in the case of no checks is implicitly passing.

Reply:
ah right.